docs/reference/platforms.rst

```diff
@@ -0,0 +1,215 @@
+Platforms
+=========
+
+Platforms
+---------
+
+The ``platforms`` keyword in a ``charmcraft.yaml`` determines where charms are
+built and where they run.
+
+For more information on how to build charms for specific bases and
+architectures, see the :doc:`Platforms how-to </howto/build-guides/platforms>`
+page.
+
+Standard notation
+~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on: [<arch>, <arch>]
```

Comment:
```suggestion
       build-on: [<arch1>, <arch2>]
```

Reply:
I like this suggestion but I realize both ways read like it needs to be exactly 2 items.

What I'm trying to convey in this reference is that `build-on` has no size constraints but `build-for` is a single element.  What do you think of?
```yaml
   platforms:
     <platform-name>:
       build-on: [<arch>, ...]
       build-for: [<arch>]
     ...
```

This also shows there can be more than 1 platform.

We could also do a style like `<arch1>, <arch2>, ..., <archX>`.

Reply:
How about a compromise of `<arch1>, ...`?

Reply:
+1 to @lengau's suggestion. You could also use letters (or nonsense words) instead of numbers to less strongly imply there's a sequence.

Reply:
I've adjusted this here to say "my-arch", "your-arch" and "our-arch", as well as the ellipsis. I chose the language because as far as I know it's fairly universal that those aren't ordered, but I'm leaving this open for @medubelko 's comments if that's not appropriate.

Reply:
Let's give it a try!